coalib/settings/Setting.py

```diff
@@ -93,15 +96,16 @@ class Setting(StringConverter):
     conversions into common data types.
     """
 
+    @enforce_signature
     def __init__(self,
                  key,
```

Comment:
let's enforce this to be `str` as well ;) who knows what'll happen in the future :D